cypress/e2e/tests/migration/applicationinventory/applications/language_discovery.test.ts

```diff
@@ -45,12 +63,69 @@ describe(["@tier2"], "Test if application language is discovered and tagged corr
         });
     });
 
+    it("Verify application discovery logic for missing, and valid credentials", () => {
+        // Automates TCs MTA-712
+
+        const application = new Application(
+            getRandomApplicationData("application discovery logic", {
+                sourceData: {
+                    repoType: "Git",
+                    sourceRepo: "https://github.com/konveyor/tackle-testapp",
+                },
+            })
+        );
+
+        application.create();
+
+        TaskManager.verifyTaskStatus(
+            application.name,
+            TaskKind.languageDiscovery,
+            TaskStatus.failed
+        );
+        TaskManager.verifyTaskStatus(application.name, TaskKind.techDiscovery, TaskStatus.canceled);
+
+        const defaultScCreds = new CredentialsSourceControlUsername(
+            getRandomCredentialsData(
+                CredentialType.sourceControl,
+                UserCredentials.usernamePassword,
+                true,
+                null,
+                true
+            )
+        );
+        credentialsList.push(defaultScCreds);
+        defaultScCreds.create();
+        defaultScCreds.verifyDefaultCredentialIcon();
+
+        const mavenDefaultCred = new CredentialsMaven(
+            getRandomCredentialsData(CredentialType.maven, null, false, null, true)
+        );
+
+        mavenDefaultCred.create();
+        mavenDefaultCred.verifyDefaultCredentialIcon();
+        credentialsList.push(mavenDefaultCred);
+
+        TaskManager.verifyTaskStatus(
+            application.name,
+            TaskKind.languageDiscovery,
+            TaskStatus.succeeded
+        );
+        TaskManager.verifyTaskStatus(
+            application.name,
+            TaskKind.techDiscovery,
+            TaskStatus.succeeded
+        );
+    });
+
     afterEach("Persist session", function () {
         Application.open(true);
     });
 
     after("Perform test data clean up", function () {
         deleteByList(applicationList);
+        deleteApplicationTableRows();
```

Comment:
```suggestion
        deleteApplicationTableRows();
```

No need to have both methods